File: skeleton/src/main/java/me/shkschneider/skeleton/helper/BundleHelper.java
package me.shkschneider.skeleton.helper;

import android.os.Bundle;
import android.support.annotation.NonNull;

import java.io.Serializable;

public class BundleHelper {

    public static Bundle pack(@NonNull final String key, final Serializable serializable) {
        final Bundle bundle = new Bundle();
        if (serializable != null) {
            bundle.putSerializable(key, serializable);
        }
        return bundle;
    }

    public static Object unpack(@NonNull final String key, final Bundle bundle) {
        if (bundle == null) {
            LogHelper.warning("Bundle was NULL");
            return null;
        }
        if (! bundle.containsKey(key)) {
            LogHelper.warning("Bundle has no such key");
            return null;
        }

        return bundle.getSerializable(key);
    }

}

File: core/src/main/kotlin/me/shkschneider/skeleton/helper/BundleHelper.kt
package me.shkschneider.skeleton.helper

import android.os.Bundle
import me.shkschneider.skeleton.helperx.Logger

import java.io.Serializable

object BundleHelper {

    // TODO @Parcelize once stable
    fun pack(bundle: Bundle, key: String, serializable: Serializable?): Bundle {
        bundle.putSerializable(key, serializable)
        return bundle
    }

    // TODO @Parcelize once stable
    fun pack(key: String, serializable: Serializable?): Bundle {
        val bundle = Bundle()
        return pack(bundle, key, serializable)
    }

    // TODO @Parcelize once stable
    fun unpack(bundle: Bundle, key: String): Serializable? {
        bundle.getSerializable(key)?.let {
            return true
        }
        Logger.warning("Bundle has no such key")
        return null
    }

}
